bindgen/ir/LocationManager.cpp

```diff
@@ -99,40 +60,36 @@ bool LocationManager::isImported(const Location &location) const {
     if (location.getPath().empty()) {
         return false;
     }
-    json headerEntry = getHeaderEntry(location);
-    return !headerEntry.empty();
+    return getHeaderEntryIndex(location) != headerEntries.size();
 }
 
-json LocationManager::getHeaderEntry(const Location &location) const {
-    for (auto it = config.begin(); it != config.end(); ++it) {
-        std::string pathToHeader = it.key();
-        if (startsWith(pathToHeader, "/")) {
-            /* full path */
-            if (location.getPath() == pathToHeader) {
-                return it.value();
-            }
-        } else if (endsWith(location.getPath(), "/" + pathToHeader)) {
-            return it.value();
-        }
-    }
-    return json::object();
+std::vector<HeaderEntry>::size_type
+LocationManager::getHeaderEntryIndex(const Location &location) const {
+    auto isHeader = [&](const HeaderEntry &headerEntry) {
+        if (startsWith(headerEntry.path, "/") &&
+            location.getPath() == headerEntry.path)
+            return true;
+        return endsWith(location.getPath(), "/" + headerEntry.path);
+    };
+    auto it =
+        std::find_if(headerEntries.begin(), headerEntries.end(), isHeader);
+    return it - headerEntries.begin();
```

Comment:
On my machine it does not compile without cast 
```c
return static_cast<std::vector<HeaderEntry>::size_type>(
        it - headerEntries.begin());
```

Reply:
Thanks, fixed.